### PR TITLE
add snyk and update to defender atp

### PR DESCRIPTION
**Pull request by lindambrown**

### CODE REVIEW

File: lib/kdi/kdi_helpers.rb [hunk 1]
```diff
@@ -85,14 +85,43 @@ def create_kdi_asset(asset_hash)
...
+	  def create_kdi_asset_vuln(asset_hash, vuln_hash, match_key=nil)
 	    kdi_initialize unless @assets
 
 	    # check to make sure it doesnt exist
+	    if match_key.nil? then
 			a = @assets.select{|a| uniq(a) == uniq(asset_hash) }.first
-			
+		else
+			case match_key
+	      		when "ip_address"
```
Comment (jcran) [inline lib/kdi/kdi_helpers.rb:98]: why split all these out vs doing

a = @assets.select{|a| a[match_key] == asset_hash.fetch(match_key)}.first

File: tasks/ms_defender_atp/lib/ms_defender_atp_helper.rb [hunk 2]
```diff
@@ -17,14 +17,13 @@ def atp_get_machines(token)
...
   end
   
-  def atp_get_vulns(token,machine_id)
-    print "Getting vulns for machine #{machine_id}"
-    atp_query_api = "https://api.securitycenter.microsoft.com/"
+  def atp_get_vulns(token,atp_query_api)
+    print "Getting vulns"
     headers = {'content-type' => 'application/json', 'accept' => 'application/json', 'Authorization' => "Bearer #{token}", 'accept-encoding' => 'identity'}
-    url =  "#{atp_query_api}/api/machines/#{machine_id}/vulnerabilities"
+    url =  "#{atp_query_api}/api/vulnerabilities/machinesVulnerabilities?skip=10000"
```
Comment (jcran) [inline tasks/ms_defender_atp/lib/ms_defender_atp_helper.rb:26]: what does the skip parameter do? Should this be passed to the function so we can paginate?

File: tasks/ms_defender_atp/ms_defender_atp.rb [hunk 3]
```diff
@@ -1,40 +1,50 @@
...
     {
-      id: "microsoft_atp",
-      name: "Microsoft ATP",
-      description: "Pulls assets and vulnerabilitiies from Microsoft ATP",
+      id: "ms_defender_atp",
+      name: "MS Defender ATP",
+      description: "Pulls assets and vulnerabilitiies from Microsoft Defenders ATP",
       options: [
-        {:name => "atp_tenant_id", 
-          :type => "strign",
```
Comment (jcran) [inline tasks/ms_defender_atp/ms_defender_atp.rb:16]: good catch!

File: tasks/ms_defender_atp/lib/ms_defender_atp_helper.rb (hunk 2 above)
Comment (lindambrown) [reply tasks/ms_defender_atp/lib/ms_defender_atp_helper.rb:26]: They don't have pagination, this is what they (microsoft) recommended I use.

File: lib/kdi/kdi_helpers.rb (hunk 1 above)
Comment (lindambrown) [reply lib/kdi/kdi_helpers.rb]: duh thanks!

**Pull request merged**